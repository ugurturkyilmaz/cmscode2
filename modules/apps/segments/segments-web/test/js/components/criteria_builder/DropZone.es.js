/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {cleanup, render} from '@testing-library/react';
import React from 'react';

import DropZone from '../../../../src/main/resources/META-INF/resources/js/components/criteria_builder/DropZone.es';

const connectDnd = jest.fn((element) => element);

describe('DropZone', () => {
	afterEach(cleanup);

	it('renders', () => {
		const OriginalDropZone = DropZone.DecoratedComponent;

		const {asFragment} = render(
			<OriginalDropZone
				connectDropTarget={connectDnd}
				dropIndex={0}
				groupId="group_01"
				onCriterionAdd={jest.fn()}
				onMove={jest.fn()}
				propertyKey="user"
			/>
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
