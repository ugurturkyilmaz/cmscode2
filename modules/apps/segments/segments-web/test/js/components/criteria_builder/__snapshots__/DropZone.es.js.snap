// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropZone renders 1`] = `
<DocumentFragment>
  <div
    class="drop-zone-root"
  >
    <div
      class="drop-zone-target"
    />
  </div>
</DocumentFragment>
`;
